python/cuml/tests/test_linear_model.py

```diff
@@ -204,6 +205,7 @@ def test_linear_regression_single_column():
         model.fit(cp.random.rand(46341), cp.random.rand(46341))
 
 
+@pytest.mark.xfail(reason="https://github.com/rapidsai/cuml/issues/4963")
```

Comment:
When I suggested the xfail originally, I was thinking that we would do a more targeted xfail for e.g. just the Hypothesis-generated random tests or just the general category of failing examples. I'm realizing now that that's probably fairly difficult to implement, although we will want a follow-up at some point to disable the Hypothesis-generated random examples.

For now, to keep this test at least semi-useful in CI, could we either temporarily pull off Hypothesis generation or duplicate the test such that we still have the original tests running on the non-Hypothesis-generated inputs?

My fault for not thinking this through earlier!

Reply:
I've chosen to take a hybrid approach and identified strategy parameters for which the test will pass and created a 
second duplicated test that is defined with fewer assumptions and marked as xfail. In this way we are testing on the original inputs, and in addition on _a_ hypothesis strategy, and make clear that there is work to be done here.

This is maybe not the most elegant solution, but it would allow us to merge this PR while maintaining the original tests.